packages/service-library/src/servicelib/redis/_project_lock.py

```diff
@@ -59,17 +62,20 @@ async def _wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
             )
             async def _exclusive_func(*args, **kwargs) -> R:
                 if notification_cb is not None:
-                    await notification_cb()
+                    with log_catch(_logger, reraise=False):
+                        await notification_cb()
```

Comment:
yep, we do not control the nature of callbacks so they must always be handled for exceptions ... I would even add a timeout i.e. they must execute within a given time ...